File: src/main/java/br/com/newObjects/model/classes/NovoProdutoFornecedor.java
package br.com.newObjects.model.classes;

import java.util.ArrayList;
import java.util.List;
import java.util.Random;

import br.com.model.entities.classes.ProdutoFornecedor;
import br.com.model.entities.classes.Usuario.Fornecedor;

public class NovoProdutoFornecedor {

    public static List<ProdutoFornecedor> novaListaProdutosFornecedor(Fornecedor fornecedor) {
        ProdutoFornecedor produtoFornecedor = new ProdutoFornecedor(NovoProduto.novoProduto(), fornecedor,
                new Random().nextFloat(), new Random().nextInt(1000));
        ProdutoFornecedor produtoFornecedor2 = new ProdutoFornecedor(NovoProduto.novoProduto2(), fornecedor,
                new Random().nextFloat(), new Random().nextInt(1000));

        List<ProdutoFornecedor> listaProdutoFornecedor = new ArrayList<ProdutoFornecedor>();
        listaProdutoFornecedor.add(produtoFornecedor);
        listaProdutoFornecedor.add(produtoFornecedor2);

        return listaProdutoFornecedor;
    }
}

File: src/test/java/br/com/newObjects/model/NovoProdutoFornecedor.kt
package br.com.newObjects.model

import br.com.model.ProdutoFornecedor
import br.com.newObjects.model.usuario.NovoFornecedor
import java.util.*

object NovoProdutoFornecedor {
    fun novaListaProdutosFornecedor(): List<ProdutoFornecedor> {
        val fornecedor = NovoFornecedor.novoFornecedor()
        val produtoFornecedor = ProdutoFornecedor(
            NovoProduto.novoProduto(),
            fornecedor,
            Random().nextInt(),
            Random().nextInt(1000).toFloat()
        )
        val produtoFornecedor2 = ProdutoFornecedor(
            NovoProduto.novoProduto2(), fornecedor, Random().nextInt(), Random().nextInt(1000).toFloat()
        )
        val listaProdutoFornecedor: MutableList<ProdutoFornecedor> = ArrayList()
        listaProdutoFornecedor.add(produtoFornecedor)
        listaProdutoFornecedor.add(produtoFornecedor2)
        return listaProdutoFornecedor
    }

    @JvmStatic
    fun novoProdutoFornecedor(): ProdutoFornecedor {
        val fornecedor = NovoFornecedor.novoFornecedor()
        return ProdutoFornecedor(
            NovoProduto.novoProduto(), fornecedor, Random().nextInt(), Random().nextInt(1000).toFloat()
        )
    }
}
